### ISSUE TITLE
Unable to perform argument verification on a Func<>

### ISSUE THREAD

**Issue by mu88** [FakeItEasy/FakeItEasy#1816]
I'm using `FakeItEasy 6.0.0` within a .NET Core 3.1 test assembly. Please see the following test code:

```c#
using System;
using System.Collections.Generic;
using FakeItEasy;
using Xunit;

namespace temp
{
    public class UnitTest1
    {
        [Fact]
        public void Test1()
        {
            // Arrange
            var dependencyFake = A.Fake<IDependency>();
            var testee = new Testee(dependencyFake);

            // Act
            testee.DoSomething(GetSomeNumbers);

            // Assert
            A.CallTo(() => dependencyFake.DoSomethingElse(GetSomeNumbers, x => x.Count)).MustHaveHappenedOnceExactly();
        }

        private List<int> GetSomeNumbers() => new() { 1, 2, 3 };
    }

    public class Testee
    {
        public Testee(IDependency dependency)
        {
            Dependency = dependency;
        }

        private IDependency Dependency { get; }

        public void DoSomething(Func<List<int>> func)
        {
            Dependency.DoSomethingElse(func, x => x.Count);
        }
    }

    public interface IDependency
    {
        void DoSomethingElse(Func<List<int>> func, Func<List<int>, int> sizeFunc);
    }
}
```

When running this code, it fails with the following message:

```
FakeItEasy.ExpectationException


  Assertion failed for the following call:
    temp.IDependency.DoSomethingElse(func: System.Func`1[System.Collections.Generic.List`1[System.Int32]], sizeFunc: System.Func`2[System.Collections.Generic.List`1[System.Int32],System.Int32])
  Expected to find it once exactly but didn't find it among the calls:
    1: temp.IDependency.DoSomethingElse(func: System.Func`1[System.Collections.Generic.List`1[System.Int32]], sizeFunc: System.Func`2[System.Collections.Generic.List`1[System.Int32],System.Int32])


   at FakeItEasy.Core.FakeAsserter.AssertWasCalled(Func`2 callPredicate, Action`1 callDescriber, CallCountConstraint callCountConstraint) in C:\projects\fakeiteasy\src\FakeItEasy\Core\FakeAsserter.cs:line 40
   at FakeItEasy.Configuration.RuleBuilder.MustHaveHappened(CallCountConstraint callCountConstraint) in C:\projects\fakeiteasy\src\FakeItEasy\Configuration\RuleBuilder.cs:line 171
   at FakeItEasy.Configuration.RuleBuilder.MustHaveHappened(Int32 numberOfTimes, Times timesOption) in C:\projects\fakeiteasy\src\FakeItEasy\Configuration\RuleBuilder.cs:line 148
   at FakeItEasy.AssertConfigurationExtensions.MustHaveHappenedOnceExactly(IAssertConfiguration configuration) in C:\projects\fakeiteasy\src\FakeItEasy\AssertConfigurationExtensions.cs:line 42
   at temp.UnitTest1.Test1() in C:\Users\me\Documents\Development\temp\UnitTest1.cs:line 21
```

I assume it is related to the fact that `x => x.Count` are two different instances. But how can I work around this without using `A<Func<List<int>, int>>.Ignored`?

**Comment by blairconrad** [comment]
Hey, @mu88. Sorry for your pain. I think your assessment is accurate. The `x => x.Count` lambdas are two separate instances, and do not compare as equals. Sadly, there is no easy workaround. The response to this problem that's received the most work and attention is the first and last part of my [answer to the StackOverflow question "How to test for a Match with FakeItEasy on a predicate call?"](https://stackoverflow.com/a/21444621/1199).

Basically testing lambdas for equality is hard, but once you can figure out a mechanism for determining whether your two `x => x.Count` arguments are the same (VasillisP does have a solution pasted at the same StackOverflow question, and I've relied on capturing the lambda and interrogating it in the past), you should be able use it just as in any other FakeItEasy validation step.

**Comment by mu88** [comment]
Thank you @blairconrad ! I think we can close the issue because you gave me a good explanation and further stuff for reading.
